packages/dds/tree/src/test/simple-tree/arrayNode.spec.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import {
+	ImplicitAllowedTypes,
+	InsertableTreeFieldFromImplicitField,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+	NodeKind,
+	SchemaFactory,
+	TreeArrayNode,
+	TreeFieldFromImplicitField,
+	TreeNodeSchema,
+} from "../../simple-tree/index.js";
+
+import { hydrate } from "./utils.js";
+import { Mutable } from "../../util/index.js";
+
+type ArrayNodeSchema<T extends ImplicitAllowedTypes> = TreeNodeSchema<
```

Comment:
Personally I'd put this in the actual arrayNode source file. While we might not use it on schema factory (where its public), it might be nice to be able to use it in the implementation logic. I did this for ObjectNode in my last refactor.

Reply:
That's a good idea - except that this one is specifically a "stripped down" version which doesn't have any of the string/naming/scope generic parameters. I coudn't figure out how to get hydrate to compile if it does have those. Again, maybe you can help me fix it to work with the existing type, in which case we could move it and share the code.